src/app/games/mancala/common/MancalaComponent.ts

```diff
@@ -156,13 +160,38 @@ export abstract class MancalaComponent<R extends MancalaRules>
         }
     }
 
-    protected async showSimpleDistribution(distribution: MancalaDistribution): Promise<MancalaDistributionResult> {
+    private async isDistributionCausingIllegalMove(x: number, config: MGPOptional<MancalaConfig>)
+    : Promise<MGPValidation>
+    {
+        const distributionResult: MancalaDistributionResult =
+            await this.showSimpleDistribution(MancalaDistribution.of(x), false);
+        if (distributionResult.endsUpInStore &&
+            config.get().mustContinueDistributionAfterStore)
```

Comment:
faudrait choisir si on prends des config en paramètre ou si on fait this.getConfig()

Reply:
this.getConfig() then !